datacube/utils/geometry/_base.py

```diff
@@ -233,6 +233,20 @@ def __eq__(self, other: SomeCRS) -> bool:
     def __ne__(self, other) -> bool:
         return not (self == other)
 
+    @property
```

Comment:
If we are going to expose this, why not get rid of `_crs`.

Reply:
Do you mean rename `_crs`  into `proj`?

This exposes private variable `_crs` in read-only fashion, just having it renamed would allow accidental assignment without updating other state variables. Honestly I would do the same `crs_str`, except I'm not sure if any code depends on that "property".

Reply:
Yes, that's what I meant, but your justification is sound. I am with you on `crs_str`, and IMHO we should make it private regardless.